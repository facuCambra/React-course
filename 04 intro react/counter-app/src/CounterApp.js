import React , {useState} from 'react';
import PropTypes from 'prop-types';



const CounterApp = ({value=10}) => {
    
    const [counter, setCounter ] = useState(0);

    console.log(counter); // retorna un valor y una funcion


    const handleAdd = (e)=>{
        setCounter( counter+1 );
        //setCounter( (c) => c + 1 ); es igual
    }
    const handleReset = ()=>{
        setCounter( value );
    }
    const handleOne =()=>{
        setCounter( counter-1 );
    }
    return(
        <>
            <h1> CounterApp</h1>
            <h2> {counter}</h2>
            <button onClick={ (e)=> {handleAdd(e)} }>++</button> 
            <button onClick={ ()=> {handleReset()}}> Reset </button>
            <button onClick={ ()=> {handleOne()}}> -1 </button>
            
        </>
    )   //<button onClick={ ()=> {setCounter(counter-1)}}> -1 </button>
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;

// racf shortcut

/* Es lo mismo que hacer
    const handleAdd = (e)=>{
        console.log(e);
    }
    return(
        <>
            <h1> CounterApp</h1>
            <h2> {value}</h2>
            <button onClick={ handleAdd }>++</button>  <----
        </>
    )   //e es el evento click
    */