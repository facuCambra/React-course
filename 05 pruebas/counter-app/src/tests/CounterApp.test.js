import CounterApp from "../CounterApp";
import React from 'react';
import '@testing-library/jest-dom' //para las ayudas
import {shallow} from 'enzyme';


describe ('Pruebas en CounterApp', ()=>{
    const value=100;
    let wrapper = shallow(<CounterApp value={value}/>);

    beforeEach( () => {
            wrapper = shallow(<CounterApp value={value}/>);
        }
    );


    test('Checkeo renderizado de CounterApp', () => { 
        //const value=10;
      //  const wrapper = shallow(<CounterApp value={value}/>); //permite simular varias cosas
        
        expect( wrapper).toMatchSnapshot(); //el snapshot
     })

     test('Valor por defecto sea 100', () => { 
       // const value=100;
        //const wrapper = shallow(<CounterApp value={value}/>); //permite simular varias cosas
        
        const defaultValue= wrapper.find('h2').text().trim();

        expect(defaultValue).toBe('100');
     })

     test('debe incrementar contador al hacer click en +1', ()=>{

       wrapper.find('button').at(0).simulate('click'); //primer boton
       const counterText = wrapper.find('h2').text().trim();
       expect(counterText).toBe('101');
     })

     test('debe dencrementar contador al hacer click en --', ()=>{

        wrapper.find('button').at(2).simulate('click'); //primer boton
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('99');  
      })

      test('Debe setear el valor x defecto al darle a reset', () => { 
          
        wrapper.find('button').at(2).simulate('click'); 
        wrapper.find('button').at(2).simulate('click'); 
        
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(value.toString());

       })

})