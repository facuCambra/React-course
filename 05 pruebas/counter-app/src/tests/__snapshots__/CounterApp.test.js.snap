// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Checkeo renderizado de CounterApp 1`] = `
<Fragment>
  <h1>
     CounterApp
  </h1>
  <h2>
     
    100
  </h2>
  <button
    onClick={[Function]}
  >
    ++
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
